### ISSUE TITLE
return promises?

### ISSUE THREAD

**Issue by pablomaurer** [alykoshin/nssm#2]
When creating a service I will need `set` around 10-15 times

``` js
nssm.set('Start', 'manual', function(error, result) {
  if (error) {
    console.log('*** error:', error, ' stderr:', result);
    return;
  }
  console.log('*** stdout: \'' + result + '\'');
});

// vs

nssm.set('Start', 'manual')
  .then(nssm.set('option', 'value'))
  .then(nssm.set('option', 'value'))
  .catch(function (err) {
    console.error(err);
});
```

In PouchDB you can even use promises, callbacks or this new async stuff. So everybody can use what he wants.. but i think promises is at the current time to most way to go.

But anyway, thank you very much for sharing!

**Comment by alykoshin** [comment]
I like your idea.
Give me some time...

**Comment by alykoshin** [comment]
Published updated version.
You may use now following syntax:

``` js
nssm.set('start', 'manual')
  .then(function(stdout) {
    return nssm.get('start')
  })
  .then(function(stdout) {
    return nssm.start()
  })
  .then(function(stdout) {
    return nssm.stop()
  })
  .then(function(stdout) {
    console.log('DONE');
  })
  .catch(function(error) {
    console.log('ERROR:', error);
  })
  ;
```

Please, test.

**Comment by jdalrymple** [comment]
Fails when attempting to use .delay functionality from the Bluebird promise API. Possibly allow for the ability to choose the promise library used? or use bluebird by default.

**Comment by alykoshin** [comment]
It uses global `Promise` constructor: https://github.com/alykoshin/nssm/blob/master/lib/index.js#L64
Can you try to set it to Bluebird, like following: `Promise = require('bluebird');` ?
Not tested this line though...

**Comment by alykoshin** [comment]
Ok, I see your PR.
But I'll move the ability to set the library to `options` object in constructor, like following:
```js
var nssm = Nssm('AeLookupSvc', { nssmExe: 'nssm.exe', Promise: require('bluebird') });
```
Ok?

**Comment by jdalrymple** [comment]
Makes sense! Could we also move the configuration up to the main require command, instead of on each service instance ie

Const nssm = require('nssm)(options)

Var myService = nssm(service name)


via Newton Mail [https://cloudmagic.com/k/d/mailapp?ct=pa&cv=9.2.13&pv=6.0.1&source=email_footer_2] On Tue, Nov 29, 2016 at 11:43am, Alexander < notifications@github.com [notifications@github.com] > wrote:
Ok, I see your PR.
But I'll move the ability to set the library to options object in constructor, like following:

var nssm =Nssm('AeLookupSvc', { nssmExe:'nssm.exe', Promise:require('bluebird') });

Ok?

**Comment by jdalrymple** [comment]
Done! Check the PR
